<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <title>How to run the program</title>
  </head>
  <body>
    <big><big><big>How to run MOOSE<sup>2</sup></big></big></big><br>
    <br>
    <br>
    <b>Building the software.</b><br>
    <br>
    MOOSE<sup>2</sup> is written in C++ and compiles and runs on Linux
    and MacOS X and requires the gcc compiler. On the Mac, you will need
    to install XCode on later OS version to get gcc.<br>
    <br>
    On the command line, type <br>
    <br>
    &gt; make<br>
    <br>
    to build the executables. To test whether the build was successful,
    run<br>
    <br>
    &gt; ./test_it<br>
    <br>
    to run the software on data that comes with the repository. Note any
    errors.<br>
    <br>
    <br>
    <b>Input files/summary</b><br>
    <br>
    MOOSE<sup>2</sup> requires two files:<br>
    <br>
    <u>1. Expression values</u>: these can either be raw read counts or
    normalized data, listed in a single file, one line per gene or
    transcript, e.g.:<br>
    <br>
    <font face="Courier New, Courier, monospace"><font face="Courier
        New, Courier, monospace">Gene
        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
        ex1&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ex2&nbsp;&nbsp;&nbsp;
        ex3&nbsp;&nbsp;&nbsp;&nbsp; ex7&nbsp;&nbsp;&nbsp;&nbsp;
        ex9&nbsp;&nbsp;&nbsp;&nbsp; ex8&nbsp;&nbsp;&nbsp;&nbsp;
        ex4&nbsp;&nbsp;&nbsp;&nbsp; ex5&nbsp;&nbsp;&nbsp;&nbsp; ex6<br>
      </font>abgT.t01&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 49.6757
      48.1321 34.522&nbsp; 33.4765 40.3088 49.9477 40.4558 53.1916
      42.0341 1527<br>
      abrB.t01&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 51.072&nbsp;
      47.2778 29.0407 30.6346 33.9294 36.2579 39.7088 45.3439 42.7197
      1047<br>
      ...<br>
    </font><br>
    Where the first line serves as a header naming the samples. If
    possible, also provide the transcript length in one of the columns,
    e.g. the last one. Columns listing expression values have to be
    contiguous.<br>
    <br>
    <br>
    <u>2. Replicate description</u>: this is an optional text file that
    specifies which samples are replicates, one line per replicate,
    preceded by an experiment name, e.g:<br>
    <br>
    <font face="Courier New, Courier, monospace">healthy&nbsp;&nbsp; ex1
      ex2 ex3<br>
      sick&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; ex7 ex8 ex9<br>
      recovered ex4 ex5 ex6<br>
    </font><br>
    Additional/optional files are: a set of known reference gene (one
    entry per line).<br>
    <br>
    <b>Running the program</b><br>
    <br>
    On the command line, type<br>
    <br>
    ./Normalize -r &lt;replicates&gt; -i &lt;fpkm_data&gt; -f &lt;first
    column containing fpkm data&gt; -l &lt;last column containing fpkm
    data&gt; -w &lt;list of reference genes&gt; -col &lt;length
    column&gt; -a &lt;average read or fragment count over all
    samples&gt;<br>
    <br>
    IMPORTANT: If using read counts instead of FPKM/RPKM values, add the
    option -counts.<br>
    <br>
    <b>************** NOTE: column indices are 0-based!!</b>
    ****************<br>
    <br>
    with the available arguments:<br>
    <br>
    <tt>-i&lt;string&gt; : input file (all expression values) </tt><tt><br>
    </tt><tt> -r&lt;string&gt; : replicates file (def=, one line per
      experiment, name followed by sample ids)</tt><tt><br>
    </tt><tt> -counts&lt;bool&gt; : process read COUNTS, not FPKMs
      (def=0)</tt><tt><br>
    </tt><tt> -a&lt;string&gt; : average read counts per sample (def=0,
      needed for correcting significance for lowly expressed
      trsnacripts)</tt><tt><br>
    </tt><tt> -col&lt;string&gt; : column that specifies the transcript
      length (def=0, needed if RPKM/FPKM values are specified to
      estimate counts)</tt><tt><br>
    </tt><tt> -w&lt;string&gt; : waypoint gene file (def=, one gene per
      line)</tt><tt><br>
    </tt><tt> -f&lt;string&gt; : first column with data (0-based)
      (def=1)</tt><tt><br>
    </tt><tt> -l&lt;string&gt; : last column with data (0-based) (def=0)</tt><tt><br>
    </tt><tt> -p&lt;string&gt; : penalty for HMM (decrease to get more
      genes) (def=) /* this is the HMM parameter</tt><tt><i> h</i></tt><tt>,
      which defaults to 5. */</tt><tt><br>
    </tt><tt> -rw&lt;string&gt; : reward to pick up genes (def=) /* this
      is the HMM parameter </tt><tt><i>m</i></tt><tt>, which defaults
      to 4. */</tt><tt><br>
    </tt><tt> -linear&lt;bool&gt; : uses a linear model (def=0)</tt><tt><br>
    </tt><tt> -force&lt;bool&gt; : forces a polynomial fit (not
      recommended) (def=0)</tt><br>
    <br>
    <br>
    The parameter -i specifies the input matrix with (a) a header
    providing the sample description; (b) a row per gene or isoform
    specifying either read or pair counts, or FPKM/RPKM normalized
    values, where the first element specifies the gene name or
    identifier.<br>
    <br>
    Data can be provided in any consecutive set of columns, which need
    to be specified via the -f (first column containing expression
    values) and -l (last column containing expression values)
    parameters, e.g. a file with 9 samples, in columns 1 through 9: <br>
    <br>
    <tt>gene&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;

      BB9 BB10 BB17 BB19 BB20 BB21 BB11 BB12 BB18 average</tt><tt><br>
    </tt><tt> aaeA.t01&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 45.6132
      57.7381 35.4817 71.4413 75.5846 59.6099 93.7032 78.2685 75.4577
      933&nbsp;&nbsp;&nbsp;&nbsp; 65.8776</tt><tt><br>
    </tt><tt> aaeB.t01&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 48.4527
      42.2968 41.1391 44.0555 47.6292 51.5953 56.3065 69.1718 58.5695
      1968&nbsp;&nbsp;&nbsp; 51.024</tt><br>
    <br>
    is provided via -f 1 -l 9. If the transcript or gene lengths is part
    of this file, optionally specify via the -col option (this is for QA
    only, as are the -counts and -a options).<br>
    <br>
    A replicate file, which is required for QA only, can be specified in
    the format:<br>
    <br>
    condition1 sampleA sampleB sampleC<br>
    condition2 sampleD sampleE<br>
    &nbsp; <br>
    where each line specifies one condition, followed by the samples
    obtained from that condition.<br>
    <br>
    A waypoint file containing confirmed invariant genes can be provided
    via the -w option as a file containing one gene name per line. Note
    that the gene names have to match exactly the names or ids in the
    data file.&nbsp; <br>
    <br>
    For an example, see the script ./test_it that comes with the
    repository.<br>
    <br>
    <br>
    <b>Output files.</b><br>
    <br>
    MOOSE<sup>2</sup> produces several output files:<br>
    <br>
    1. hmm_out: the set of genes assumed to be invariant in expression.<br>
    <br>
    2. normalized.out: the normalized expression values<br>
    <br>
    3. distribution.txt: the distribution parameters and values or
    within replicate comparisons for quality assurance purposes<br>
    <br>
    <b><br>
      Controlling the number of <i>in silico</i> invariant genes.</b><br>
    <br>
    The number of invariant gene predictions depends on a number of
    factors, including the number of samples, number of genes etc. To
    increase the number of genes, increase the HMM reward m (-p option,
    default=5) and/or decrease the HMM penalty h (-rw option,
    default=4). In the example data, the default setting produce 33
    predictions, -m 6 -h 3 results in 53 predictions etc.<br>
    <br>
    <br>
    <b>Quality assessment (optional)</b><br>
    <br>
    For quality assessment, the file distribution.txt contains both the
    observed distribution, as well as the fitted model. If you visualize
    both (note that the first line in the file contains the parameters,
    the rest lists: position, observed, fitted), you should see a
    distribution like this one (red = observed, black = model):<br>
    <br>
    <img alt="Distribution" title="Distribution" src="distrib.png"
      height="354" width="500"><br>
    <br>
    where both follow each other very closely. A discrepancy might be
    indicative of an experimental problem.<br>
    <br>
    <br>
    <br>
  </body>
</html>
